### ISSUE TITLE
Admin token vs Agent token

### ISSUE THREAD

**Issue by jsgitit** [shenanigan/fbi-demo#2]
Hello again, I'm still working through your new article and very much appreciate the experience of applying tokens and using SwaggerUI to test the APIs.  I do have a follow up concern.  I'm confused on the difference between the Admin token and Agent token.  In your blog, you mentioned: 

> "If you try to access these API’s with the Admin token you will get 403 Forbidden error. So go ahead and click on the Authorize button again and click on logout. Then, get the token from the above step and paste in the field with Bearer as a prefix i.e. Bearer .
> 
> When you access /Agent/AccessPublicFiles api you will see response 200 with message Public Files Accessed. But when you try the classified api you get 403 Forbidden error.

I received the 403 Forbidden error as mentioned.  But this is where it gets confusing... When you say, "get the token from the above step and paste it with the Bearer + key", I am effectively regenerating the token as Admin, not as Agent.  There is no Agent/Login API.  Am I supposed to have a separate Login() method for Agent?  Otherwise I will keep using Admin, correct?  I will always receive 403 when trying to run Agent APIs as Admin.

I continue to get the 403 error, even when I get the token again. I may not be following instructions very well.  Thank you for clarifying what I should do in order to get the "Public Files Accessed" message.

**Comment by shenanigan** [comment]
Hi Jon, 
The /Admin/Login api generates the bearer token for Admin, while the /Admin/GenerateBadge api generates the bearer token for Agent. 

So if you want to access any api that is authorised and falls under AdminController use the Admin's bearer token. 
Similarly, if you want to access any api that is authorised and falls under AgentController use the Agent's bearer token. 

I hope this clears. things up. I will keep this issue open till your confusion is cleared.

**Comment by jsgitit** [comment]
I see, this helps.  But more importantly, I now see the updated AgentController.cs.  Previously, the implementation was not complete and this was the primary source of my confusion.  After I updated the code, now the proper token was being used and I can get the expected output.  In the first blog, we had the following implementation for AgentController.cs:

`[HttpPost]
[Route("[action]")]
[Authorize(Policy = "Admin")]
public IActionResult GenerateBadge([FromBody] Agent Agent)
{
    // TODO: Add the agent to the database
    return new CreatedResult("Agent/1", Agent); // not the full implementation! - JS
}`

Thanks for all your help.  It's been fun to work through this article and I learned a lot!  Looking forward to more content.
